### PR TITLE
fix(azure): Support exact branch merge policies.

### PR DESCRIPTION
**Pull request by scabana**
## Changes

Renovate will now honor merge strategy policies targeting a branch  when one exists at the Azure DevOps Project level. 

## Context

This resolves the issue in this discussion: #27330

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/azure/azure-helper.spec.ts [hunk 1]
```diff
@@ -238,6 +238,33 @@ describe('modules/platform/azure/azure-helper', () => {
...
+                      repositoryId: null,
+                    },
+                  ],
+                },
+                type: {
+                  id: 'fa4e907d-c16b-4a4c-9dfa-4916e5d171ab',
+                },
+              },
+            ]),
+          }) as any,
```
Comment (viceice) [inline lib/modules/platform/azure/azure-helper.spec.ts:261]: please use our partial helper, there should be some samples already.

Comment (scabana) [reply lib/modules/platform/azure/azure-helper.spec.ts]: Is this better?

Comment (scabana) [reply lib/modules/platform/azure/azure-helper.spec.ts]: looking into the lint errors

Comment (scabana) [reply lib/modules/platform/azure/azure-helper.spec.ts]: Output of `pnpm run lint` looks good now.

**Review by viceice — APPROVED** [review]

**Pull request merged**